src/components/candevice/candevice.cpp

```diff
@@ -100,6 +128,43 @@ QJsonObject CanDevice::getConfig() const
     return {};
 }
 
+void CanDevice::setConfig(const QObject& qobject)
+{
+    Q_D(CanDevice);
+
+    for (const auto& p: getSupportedProperties())
+    {
+        QVariant v = qobject.property(p.first.toStdString().c_str());
+        if (!v.isValid() || v.type() != p.second.first)
+            continue;
+
+        d->_props[p.first] = v;
+    }
+
+    init();
+}
+
+std::shared_ptr<QObject> CanDevice::getQConfig() const
+{
+    const Q_D(CanDevice);
+
+    std::shared_ptr<QObject> q = std::make_shared<QObject>();
+
+    QStringList props;
+    for (auto& p: getSupportedProperties())
+    {
+        if (!p.second.second) // property not editable
```

Comment:
No need to `continue`, say that explicitly:

```
if (true == p.second.second)  // editable?
{ ... }
```

Reply:
It's not necessary, but the reason for such style is to minimize levels of indentation. I try to avoid having more than two levels of them.

Reply:
My intention was to expose the condition and simplify the logic here.  With `continue` logic is "reversed", i.e. code is supposed to execute certain algorithm *if* given condition is met for an item from the collection over which we iterate (that's classic `filter` algorithm in fact).

Reply:
I will change it as you wish... but the logic is not at all more complicated this way. You simply look through the properties, skip those that don't meet criteria and apply changes to those that meet them.